tests/dtm0/test_single_process_restart.py

```diff
@@ -0,0 +1,169 @@
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+#
+# For any questions about this software or licensing,
+# please email opensource@seagate.com or cortx-questions@seagate.com.
+#
+
+"""
+Test suite for testing Single Process Restart with DTM enabled.
+"""
+import logging
+import multiprocessing
+import os
+
+import pytest
+
+from commons import configmanager
+from commons.constants import K8S_SCRIPTS_PATH, POD_NAME_PREFIX, MOTR_CONTAINER_PREFIX
+from commons.helpers.health_helper import Health
+from commons.helpers.pods_helper import LogicalNode
+from commons.params import LATEST_LOG_FOLDER
+from commons.utils import support_bundle_utils, assert_utils
+from config import CMN_CFG
+from conftest import LOG_DIR
+from libs.dtm0.dtm0_lib import DTM0TestLib
+
+
+class TestSingleProcessRestart:
+    """Test Class for Single Process Restart."""
+
+    @classmethod
+    def setup_class(cls):
+        """Setup class."""
+
+        cls.log = logging.getLogger(__name__)
+        cls.num_nodes = len(CMN_CFG["nodes"])
+        cls.master_node_list = []
+        cls.worker_node_list = []
+        for node in CMN_CFG['nodes']:
+            node_obj = LogicalNode(hostname=node["hostname"],
+                                   username=node["username"],
+                                   password=node["password"])
+            if node["node_type"].lower() == "master":
+                cls.master_node_list.append(node_obj)
+            else:
+                cls.worker_node_list.append(node_obj)
+        cls.setup_type = CMN_CFG["setup_type"]
+        cls.test_completed = False
+        cls.health_obj = Health(cls.master_node_list[0].hostname,
+                                cls.master_node_list[0].username,
+                                cls.master_node_list[0].password)
+        cls.test_cfg = configmanager.get_config_wrapper(fpath="config/test_dtm0_config.yaml")
+        cls.m0d_process = 'm0d'
+        cls.restart_wait_time_secs = 10
+        cls.dtm0_obj = DTM0TestLib()
+
+    def teardown_method(self):
+        """Teardown class method."""
+        if not self.test_completed:
+            self.log.info("Test Failure observed, collecting support bundle")
+            path = os.path.join(LOG_DIR, LATEST_LOG_FOLDER)
+            resp = support_bundle_utils.collect_support_bundle_k8s(local_dir_path=path,
+                                                                   scripts_path=K8S_SCRIPTS_PATH)
+            assert_utils.assert_true(resp)
+        # TODO : Redeploy setup after test completion.
+
+    @pytest.mark.lc
+    @pytest.mark.dtm0
+    @pytest.mark.tags("TEST-41204")
+    def test_read_during_m0d_restart(self):
+        """Verify READ during m0d restart using pkill -9."""
```

Comment:
From test design perspective is "-9" a mandatory step?

Reply:
Removed from tests.